chrome/content/zotero/xpcom/style.js

```diff
@@ -428,71 +428,94 @@ Zotero.Styles = new function() {
 	
 	/**
 	 * Populate menulist with locales
+	 * 
+	 * @param {xul:menulist} menulist
 	 */
-	this.populateLocaleList = function(menulist, prefLocale) {
+	this.populateLocaleList = function(menulist) {
 		if(!_initialized) this.init();
 		
 		// Reset menulist
 		menulist.selectedItem = null;
 		menulist.removeAllItems();
 		
-		var doc = menulist.ownerDocument;
-		var popup = doc.createElement('menupopup');
-		menulist.appendChild(popup);
+		let fallbackLocale = Zotero.Styles.primaryDialects[Zotero.locale]
+			|| Zotero.locale;
 		
-		var desiredLocale = "";
-		var fallbackLocale = Zotero.locale;
+		let menuLocales = Zotero.Utilities.deepCopy(Zotero.Styles.locales);
+		let menuLocalesKeys = Object.keys(menuLocales).sort();
 		
-		// Primary dialect conversion (e.g. "en" to "en-US")
-		if (Zotero.Styles.primaryDialects[prefLocale] !== undefined) {
-			prefLocale = Zotero.Styles.primaryDialects[prefLocale];
-		}
-		if (Zotero.Styles.primaryDialects[fallbackLocale] !== undefined) {
-			fallbackLocale = Zotero.Styles.primaryDialects[fallbackLocale];
+		// Make sure that client locale is always available as a choice
+		if (fallbackLocale && !(fallbackLocale in menuLocales)) {
+			menuLocales[fallbackLocale] = fallbackLocale;
+			menuLocalesKeys.unshift(fallbackLocale);
 		}
 		
-		if (prefLocale) {
-			desiredLocale = prefLocale;
-		} else {
-			desiredLocale = fallbackLocale;
+		for (let i=0; i<menuLocalesKeys.length; i++) {
+			menulist.appendItem(menuLocales[menuLocalesKeys[i]], menuLocalesKeys[i]);
+		}
+	};
+	
+	/**
+	 * Update locale list state based on style selection.
+	 *   For styles that do not define a locale, enable the list and select a
+	 *     preferred locale.
+	 *   For styles that define a locale, disable the list and select the
+	 *     specified locale. If the locale does not exist, it is added to the list.
+	 *   If null is passed instead of style, the list and its label are disabled,
+	 *    and set to blank value.
+	 * 
+	 * Note: Do not call this function synchronously immediately after
+	 *   populateLocaleList. The menulist items are added, but the values are not
+	 *   yet set.
+	 * 
+	 * @param {xul:menulist} menulist Menulist object that will be manipulated
+	 * @param {Zotero.Style} style Currently selected style
+	 * @param {String} prefLocale Preferred locale if not overridden by the style
+	 * 
+	 * @return {String} The locale that was selected
```

Comment:
Is there documentation on how to format this kind of parameter documentation (e.g. "@return {String} The locale that was selected")? Curious.